test/freenet/client/filter/M3UFilterTest.java

```diff
@@ -56,6 +59,7 @@ public void testSuiteTest() throws IOException {
                 assertTrue("Creating URI from BASE_URI " + BASE_URI + " failed", false);
             }
         }
+        assertTrue("All files should be readable, but at least " + unreadableFiles.size() + " files could not be read: " + unreadableFiles.toString(), unreadableFiles.isEmpty());
```

Comment:
No, this is getting worse all the time, don’t add more asserts, let the test crash & burn if *the test setup* fails. :)

Reply:
This gives better output if multiple files are missing. The output would be misleading if it said "one file not found" while in reality no single file can be found.

Reply:
No, we’re going down the wrong path here. Assertions and their nice messages are for the actual tests; “counted 5 but expected 8.” The test setup itself should (IMHO) throw exceptions because there’s not necessarily anything wrong with the actual code. Failed assertions are invitations for being ignored.

Reply:
I think giving more complete info wins here, but that opinion isn’t a particularly strongly held one :-). So it now throws directly when hitting the first missing file.